Add tic-tac-toe

tanishdadev opened the pull request:
# TerminalCraft Project Submission

## Project Details

1. **What is the name of your project?**
Tic-Tac-Toe
2. **How many hours did you spend developing this?**
   _(Give your best estimate or show proof using hackatime/wakatime)_
![Screenshot 2025-03-17 195101](https://github.com/user-attachments/assets/be244b9b-e708-43dc-8185-2b7870fa84a5)

3. **Briefly describe what your project is. What makes it unique?**
   _(Provide a brief description of the functionality or innovation)_
This project is tic-tac-toe made in C, where in you beat the computer to win the game. (Couldn't make it too unique because of time constraints and final exams but i spent around 3 hours of the spare time I had on this and I am still a beginner.)
4. **How many people have tested your extension?**
   _(We recommend at least 3)_
0 unfortunately
5. **On What operating systems have you tested this project?**
   _(MacOS/Linux/Windows)_
Windows
6. **Does your project include a Readme file with instructions on how to use it?**
   _(Yes/No)_
Yes
---

## Additional Comments
Yeah so I have talked to you about it that I couldn't test it on Linux due to time constraints sorry :)
(Feel free to provide any additional context or notes here)

JosiasAurel reviewed (changes requested):
2 hrs is wayyy below the 7hrs requirement. There also exist a much more complex tic tac toe game among approved projects. I suggest looking into features you could add to make yours stand out a bit more or starting a new project.

tanishdadev commented:
I have added the multiplayer feature where people can take turns and also play with the AI if they want to, and i have changed the menu a bit and kept it simple this time + changed some features and layout as well.

shubhisroking reviewed (changes requested):
there are way too many tic tac toe submissions and unfortunately your project doesn't standout

JosiasAurel commented on submissions/tic-tac-toe/tic-tac-toe.exe:
remove this file

JosiasAurel reviewed (changes requested):
How is this different from existing submissions?

tanishdadev commented:
> How is this different from existing submissions?

@JosiasAurel I haven't seen the other submissions but it's a fun game. I have added 8 levels of tic tac toe. You can play on 3x3 board or a 10x10 board also. You will get replay of what you and computer played and its written in C, which is fairly a new language to me. I contributed the max amount of time I could, during my final examinations (which are now over btw). Considering the fact that I'm still a beginner and added those features make me proud myself. Another thing, if I had known that you will review the projects and decline them because there are too many of the same (which was not written in the original message in the announcement channel) , then I wouldn't even have bothered to participate in the ysws during my final exams because I know that I wouldn't have ofc stood out due to limited amount of time available. I would request to kindly consider all the facts again and review the pr. Thanks a lot!

JosiasAurel reviewed (approved):
It should be able to handle incorrect inputs. For example, I entered `4i` by mistake in one of the inputs and it started an infinite loop.
<img width="1466" alt="image" src="https://github.com/user-attachments/assets/b57728c2-fa37-43b4-81a8-77d36e22ada4" />

other than that, this is okay! please fill [this form](https://airtable.com/appzR6MIcj5G9A2Fa/pag8jn2axMXOB2lid/form) if you haven't yet
